lectures/5-commitments.tex

```diff
@@ -0,0 +1,159 @@
+
+    \item \textit{Commit:} prover selects a value and generates a commitment to this value. This commitment is then sent to the verifier;
+
+    \item  \textit{Verify:} prover reveals the committed value and any necessary information to the verifier. 
+\end{enumerate}
+
+Properties of commitments scheme:
+\begin{enumerate}
+    \item \textit{Hiding: } verifier should not learn any information about the message given only the commitment c;
+    \item \textit{Binding: } prover could not find another message $m_1$ and open the commitment $c$ without revealing the commited message $m$.
+\end{enumerate}
+
+\subsubsection{Hash-based commitments}
+
+As the name implies, we are using a cryptographic hash function \(H\) in such scheme.
+
+\begin{enumerate}
+    \item Prover selects a message $m$ which he wants to commit:
+        $m \leftarrow \mathbb{Z}$
+
+    \item Prover samples random value $r$ (usually called blinding factor) from $\mathbb{Z}$:
+        $r \xleftarrow{R} \mathbb{Z}$
+    
+    \item Both values will be concatenated and hashed with the hash function $H$ to produce the commitment:
+        $c = H(m || r)$
+
+\end{enumerate}
+
+Commitment should be shared with a verifier. During the verification stage, prover reveals $(m, r)$ to the verifier. 
+To check the commitment, verifier computes: $c_1 = H(m || r)$.
+
+If $c_1 = c$, prover has revealed the correct pair $(m, r)$.
+
+\subsubsection{Pedersen commitments}
+
+Pedersen commitments allow us to represent arbitrarily large vectors with a single elliptic curve point, while optionally hiding any information about the vector. Pedersen commitment uses a public group $\mathbb{G}$ of order $q$ and two random public generators $G$ and $U$: $U = uG$. Secret parameter $u$ should be hidden, otherwise the $\textit{Binding}$ property of the commitment scheme will be violated.
+EC point $U$ is chosen randomly using "Nothing-up-my-sleeve" to assure no one knows the discrete logarithm of a selected point.
+
+\begin{remark}
+    \textbf{Transparent random points generation}
+
+    User can pick the publicly chosen random number (like a hash of project name, first numbers of $\pi$, etc), and hash that result to obtain another value. If that results in an $x$ value that lies on the elliptic curve, use that as the random point and hash the $(x, y)$ pair again (to obtain the next one, it needed). Otherwise, if the $x$-value does not land on the curve, increment $x$ until it does. Because the committer is not generating the points, they don`t know their discrete log. 
+\end{remark}
+
+Pedersen commitment scheme algorithm:
+\begin{enumerate}
+    \item Prover and Verifier agrees on $G$ and $U$ points
+    \item Prover selects a value $m$ to commit and a blinder factor $r$:
+            $m \leftarrow \mathbb{Z}$,
+            $r \xleftarrow{R} \mathbb{Z}$
+    \item Prover generates a commitment and sends it to the Verifier:
+            $c = mG + rU$
```

Comment:
I recommend using notation $c \gets [m]G + [r]U$, which is read as "c gets initialized as an expression..."